cleo_plugins/IniFiles/IniFiles.cpp

```diff
@@ -34,21 +35,14 @@ class IniFiles
 		0AF0=4,%4d% = get_int_from_ini_file %1s% section %2s% key %3s%
 		****************************************************************/
 	{
-		char path[MAX_PATH];
-		char sectionName[100];
-		char key[100];
-		int result;
+		auto path = OPCODE_READ_PARAM_FILEPATH();
+		char sectionName[64]; OPCODE_READ_PARAM_STRING_BUFF(sectionName, sizeof(sectionName));
+		char key[64]; OPCODE_READ_PARAM_STRING_BUFF(key, sizeof(key));
 
-		CLEO_ReadStringPointerOpcodeParam(thread, path, sizeof(path));
-		CLEO_ReadStringPointerOpcodeParam(thread, sectionName, sizeof(sectionName));
-		CLEO_ReadStringPointerOpcodeParam(thread, key, sizeof(key));
-
-		CLEO_ResolvePath(thread, path, sizeof(path)); // convert to absolute path
-
-		result = GetPrivateProfileInt(sectionName, key, 0x80000000, path);
-		CLEO_SetIntOpcodeParam(thread, result);
-		CLEO_SetThreadCondResult(thread, result != 0x80000000);
+		auto result = GetPrivateProfileInt(sectionName, key, 0x80000000, path);
 
+		OPCODE_WRITE_PARAM_INT(result);
+		OPCODE_CONDITION_RESULT(result != 0x80000000);
```

Comment:
suggestion: get rid of unnecessary write operation if result is false.

Reply:
Some mods might relay on 0x80000000 to detect the error.

Reply:
they should use if ... then

Reply:
Too late for already existing mods.
```
read int 0@
if
    0@ == 0x80000000 
then
    0@ = 4 // default value
end
```
Code like this won't work any more.